embd.go

```diff
@@ -72,20 +73,36 @@ func run() error {
 		Args:  os.Args[1:],
 		Pkg:   *pkg,
 		Files: map[string]File{},
+		Dirs:  make(map[string][]File),
 	}
 	for _, path := range flag.Args() {
-		// TODO(akavel): support directories
-		f, err := NewFile(path)
+		fi, err := os.Stat(path)
 		if err != nil {
 			return err
 		}
-		if old, exists := contents.Files[f.VarName]; exists {
-			return fmt.Errorf(
-				"generated variable name conflict: '%s' resolves to"+
-					"the same variable name %s as '%s'",
-				f.Path, f.VarName, old.Path)
+
+		if fi.IsDir() {
+			f, _ := ioutil.ReadDir(path)
+			for _, p := range f {
+				f, err := NewFile(path + "/" + p.Name())
```

Comment:
`+ "/" +` is risky here from cross-platform-ness point of view. The `path` value may have either `/` or `\` inside depending on OS. Please call filepath.ToSlash on `path` just before os.Stat.